File: PianoRisePython/app/models/usuario.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Date, DateTime, Boolean, ForeignKey, Text
from sqlalchemy.orm import relationship
from app import db


class Usuario(db.Model):
    __tablename__ = "usuario"

    id                 = Column("id", Integer, primary_key=True)
    nombre_completo    = Column("nombre_completo", String(100), nullable=False)
    correo_electronico = Column("correo_electronico", String(100), unique=True, nullable=False)
    contrasena_hash    = Column("contrasena_hash", Text, nullable=False)

    rol_id             = Column("rol_id", Integer, ForeignKey("rol.id"), nullable=False)
    rol                = relationship("Rol", back_populates="usuarios")

    telefono           = Column("telefono", String(15), nullable=True)
    fecha_registro     = Column("fecha_registro", DateTime, default=datetime.utcnow, nullable=False)
    fecha_nacimiento   = Column("fecha_nacimiento", Date, nullable=True)
    esta_activo        = Column("esta_activo", Boolean, default=True, nullable=False)

    alumno_aulas       = relationship("AlumnoAula", back_populates="alumno", lazy="dynamic")
    historiales        = relationship("Historial",   back_populates="usuario", lazy="dynamic")
    
    @property
    def is_active(self):
        return self.esta_activo
    
    def get_id(self):
        return str(self.id)
    
    @property
    def is_authenticated(self) -> bool:
        return True

    @property
    def is_anonymous(self) -> bool:
        return False

File: PianoRiseDotnet/PianoRiseApp/Models/Usuario.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PianoRiseManagement.Models
{
    [Table("usuario")]
    public class Usuario
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required(ErrorMessage = "El nombre completo es obligatorio.")]
        [MaxLength(100)]
        [RegularExpression(@"^[A-Za-zÁÉÍÓÚáéíóúÑñ\s]+$", ErrorMessage = "El nombre solo debe contener letras y espacios.")]
        [Column("nombre_completo")]
        public string NombreCompleto { get; set; } = null!;

        [Required]
        [MaxLength(100)]
        [EmailAddress]
        [Column("correo_electronico")]
        public string CorreoElectronico { get; set; } = null!;

        [Required]
        [MaxLength(500)]
        [Column("contrasena_hash")]
        public string ContrasenaHash { get; set; } = null!;

        [Column("rol_id")]
        public int RolId { get; set; }

        [ForeignKey(nameof(RolId))]
        public Rol? Rol { get; set; }

        [MaxLength(15)]
        [RegularExpression(@"^\d{9}$", ErrorMessage = "El teléfono debe tener 9 dígitos.")]
        [Column("telefono")]
        public string? Telefono { get; set; }

        [Column("fecha_registro")]
        public DateTime FechaRegistro { get; set; } = DateTime.Now;

        [DataType(DataType.Date)]
        [Column("fecha_nacimiento")]
        public DateTime? FechaNacimiento { get; set; }

        [Column("esta_activo")]
        public bool EstaActivo { get; set; } = true;

        public ICollection<Aula>? AulasDocente { get; set; }
        public ICollection<AlumnoAula>? AlumnoAulas { get; set; }
        public ICollection<Historial>? Historiales { get; set; }
    }
}
